aws-sdk-core/spec/aws/cloud_front/cookie_signer_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+module Aws
+  module CloudFront
+    describe CookieSigner do
+
+      let(:options) {
+        {
+          :key_pair_id => "CF_KEYPAIR_ID",
+          :private_key_path =>"#{File.dirname(__FILE__)}/cf_private_key.pem"
+        }
+      }
+      let(:signer) { Aws::CloudFront::CookieSigner.new(options) }
+      let(:expires) { 1357034400 } # January 1, 2013 10:00 am UTC (Unix timestamp)
+
+      describe '#signed_cookie' do
+        it 'raises error if url is invaild' do
+          expect {
+            signer.signed_cookie("what_ever_ilegal/url")
+          }.to raise_error(ArgumentError)
+        end
+        it 'can generate signed urls with custom policy' do
+          policy =  {
+            'Statement' => [
+              'Resource' => 'images/image.jpg',
+                'Condition' => {
+                'IpAddress' => {'AWS:SourceIp' => '10.52.176.0/24'},
+                'DateLessThan' => {'AWS:EpochTime' => expires}
+                }
+              ]
+          }
+          cookie = signer.signed_cookie(
+            "http://abc.cloudfront.net/images/image.jpg",
+            policy: policy.to_json
+          )
+          expect(cookie.keys).to match_array(["CloudFront-Policy", "CloudFront-Signature", "CloudFront-Key-Pair-Id"])
```

Comment:
Perhaps you are testing this as what PHP SDK does : )
I'd more like to have this to test the actual value, since we already have a mock private key there.
e.g., similar to how we test [`url_signer`](https://github.com/aws/aws-sdk-ruby/blob/master/aws-sdk-core/spec/aws/cloud_front/url_signer_spec.rb#L53), I'd like to see not only those key matches, but also their value make sense : )